homeassistant/components/becker/config_flow.py

```diff
@@ -0,0 +1,96 @@
+"""Config flow to configure Becker component."""
+import logging
+import os
+
+import serial.tools.list_ports
+import voluptuous as vol
+
+from homeassistant import config_entries
+
+from .const import CONF_DEVICE, CONF_DEVICE_PATH, DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+CONF_MANUAL_PATH = "Enter Manually"
+
+
+class BeckerConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Becker config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    def __init__(self):
+        """Initialize the Becker config flow."""
+        self._device_path = CONF_DEVICE
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow start."""
+        if self._async_current_entries():
+            return self.async_abort(reason="one_instance_only")
+
+        ports = await self.hass.async_add_executor_job(serial.tools.list_ports.comports)
+        list_of_ports = [
+            f"{p}, s/n: {p.serial_number or 'n/a'}"
+            + (f" - {p.manufacturer}" if p.manufacturer else "")
+            for p in ports
+        ]
+
+        if user_input is not None:
+            user_selection = user_input[CONF_DEVICE_PATH]
+
+            port = ports[list_of_ports.index(user_selection)]
+            dev_path = await self.hass.async_add_executor_job(
+                get_serial_by_id, port.device
+            )
+
+            self._device_path = dev_path
+            return await self.async_step_port_config()
+
+        schema = vol.Schema({vol.Required(CONF_DEVICE_PATH): vol.In(list_of_ports)})
+        return self.async_show_form(step_id="user", data_schema=schema)
+
+    async def async_step_import(self, info):
```

Comment:
We shouldn't support importing. It's a new integration and so this is not allowed.

Reply:
This is something that would still need addressing. Why can't you use the service ?

Reply:
You're right, I'll remove this part, this was basically for my own case as I was using this a custom component

Reply:
@balloob finally it was really used as the covers are setup via yaml, this import is used to init the entities in async_setup
maybe it's not the right way but I was asked so many time to change the implementation that I'm completely lost. To be honnest it's working fine for me as a custom component, it's sad but I think I'll keep it for me as it is then

Reply:
this is too bad, been waiting for this since first pushed in may. I will look in to your code nicolasberthel to see if i atleast can have it playing nice in my setup as a custom component

Reply:
> this is too bad, been waiting for this since first pushed in may. I will look in to your code nicolasberthel to see if i atleast can have it playing nice in my setup as a custom component
If you are more confident with Python that I am feel free to use my code, I can handover the integration to you ;)

Reply:
I'm definitely not confident at all. Can barely do anything